pkg/metrics/metrics.go

```diff
@@ -179,10 +180,25 @@ type SendMetricRequest struct {
 	InstallID string `json:"installId"`
 }
 
-// WriteMetric sends information about application usage. Noop if metrics
-// are opted out.
+// WriteMetric calls WriteMetricSync in a go routine.
+// It returns a closure to be run after program logic which will block until a response is returned
+// or provided context is canceled. If no provided deadline in context, defaults to 2 seconds.
+func (c *client) WriteMetric(ctx context.Context, name string, count int64) func() {
+	cancel := func() {}
+	if _, ok := ctx.Deadline(); !ok {
+		ctx, cancel = context.WithTimeout(ctx, time.Second*2)
+	}
+
+	return asyncFunctionCall(ctx, func() error {
+		defer cancel()
+		return c.WriteMetricSync(ctx, name, count)
+	})
+}
+
+// WriteMetricSync sends information about application usage. Noop if metrics
+// are opted out. Blocks until completion.
 // Accepts a context for cancellation.
-func (c *client) WriteMetric(ctx context.Context, name string, count int64) error {
+func (c *client) WriteMetricSync(ctx context.Context, name string, count int64) error {
```

Comment:
From a style perspective, it's much more common to have the default be sync, and then callers can run `go thing()` to make it async.

Reply:
Changed. I'll change the existing `updater` package to follow this pattern as well (haven't cut v1.0 yet so signature change is fair game)